pkg/storage/stores/shipper/shipper_index_client.go

```diff
@@ -54,18 +55,18 @@ type boltDBIndexClient interface {
 }
 
 type Config struct {
-	ActiveIndexDirectory     string                   `yaml:"active_index_directory"`
-	SharedStoreType          string                   `yaml:"shared_store"`
-	SharedStoreKeyPrefix     string                   `yaml:"shared_store_key_prefix"`
-	CacheLocation            string                   `yaml:"cache_location"`
-	CacheTTL                 time.Duration            `yaml:"cache_ttl"`
-	ResyncInterval           time.Duration            `yaml:"resync_interval"`
-	QueryReadyNumDays        int                      `yaml:"query_ready_num_days"`
-	IndexGatewayClientConfig IndexGatewayClientConfig `yaml:"index_gateway_client"`
-	BuildPerTenantIndex      bool                     `yaml:"build_per_tenant_index"`
-	IngesterName             string                   `yaml:"-"`
-	Mode                     int                      `yaml:"-"`
-	IngesterDBRetainPeriod   time.Duration            `yaml:"-"`
+	ActiveIndexDirectory     string              `yaml:"active_index_directory"`
+	SharedStoreType          string              `yaml:"shared_store"`
+	SharedStoreKeyPrefix     string              `yaml:"shared_store_key_prefix"`
+	CacheLocation            string              `yaml:"cache_location"`
+	CacheTTL                 time.Duration       `yaml:"cache_ttl"`
+	ResyncInterval           time.Duration       `yaml:"resync_interval"`
+	QueryReadyNumDays        int                 `yaml:"query_ready_num_days"`
+	IndexGatewayClientConfig indexgateway.Config `yaml:"index_gateway_client"`
```

Comment:
I think it would be good to keep `IndexGateway` and `IndexGatewayClient` config separate for better understanding.
You can add the ring config to just `IndexGateway.Config` and handle set a ring reference in `IndexGatewayClient.Config`.

You can see it as an example in [SchedulerWorkerConfig](https://github.com/grafana/loki/blob/main/pkg/querier/worker_service.go#L31) which is used by queriers. And the ring config is only added to [Scheduer.Config](https://github.com/grafana/loki/blob/main/pkg/scheduler/scheduler.go#L125).

Does it make sense?

Reply:
Sorry, I tried it but I have a few questions:
- Are you suggesting to have every configurable YAML definition field in `indexgateway.Config` and to have a copy of them in this `shipper_index_client.go` file but without the YAML references?
- Or are you actually suggesting splitting the YAML definition into both, `indexgateway.Config` and `shipper_index_client.go` file? If so, I found it hard to correctly register the flags.

Reply:
Sorry for the confusion. I meant splitting the yaml for index gateway client and server. Client config would stay in `pkg/storage/stores/shipper/shipper_index_client.Config` while the index gateway server config(containing just ring config) would be with `pkg/storage/stores/shipper/indexgateway`.

Sorry I gave wrong reference above for [SchedulerWorkerConfig](https://github.com/grafana/loki/blob/main/pkg/querier/worker_service.go#L31). It should be [WorkerServiceConfig](https://github.com/grafana/loki/blob/a2dd73158621441fe2d83346e792913c6770c945/pkg/querier/worker_service.go#L31).

Please let me know if you need any help here.

Reply:
Thanks for the clarification, it is clear now. I pushed a recent commit changing it, lmk if that's what you were thinking.